src/Datasource/EntityTrait.php

```diff
@@ -1091,17 +1085,12 @@ public function setInvalid(array $fields, $overwriteValue = null, $overwrite = f
      * or to be able to log it away.
      *
      * @param string $field The value to set.
-     * @param mixed|null $value The invalid value to be set for $field.
-     * @param bool $overwrite Whether or not to overwrite pre-existing values for $field.
+     * @param mixed $value The invalid value to be set for $field.
      * @return $this
      */
-    public function setInvalidField($field, $value = null, $overwrite = false)
+    public function setInvalidField($field, $value)
     {
-        if ($overwrite === true) {
-            $this->_invalid[$field] = $value;
-        } else {
-            $this->_invalid += [$field => $value];
-        }
+        $this->_invalid += [$field => $value];
```

Comment:
Same as `setInvalid()`. Now there's no way to overwrite a value at all.

This though should always overwrite the value. If you call a setter explicitly for a given field, you expect the value is set regardless.